### PR TITLE
Move Flutter SDK-related calls to FlutterSdk.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: lib/src/sdk_env.dart [hunk 1]
```diff
@@ -69,13 +62,49 @@ class DartSdk {
   }
 }
 
+class FlutterSdk {
+  String _flutterBin;
```
Comment (kevmoo) [inline lib/src/sdk_env.dart]: make this final.

File: lib/src/sdk_env.dart [hunk 2]
```diff
@@ -69,13 +62,49 @@ class DartSdk {
...
+
+  FlutterSdk({String sdkDir}) {
+    String path = _prepareSdkPath(sdkDir);
+    _flutterBin = '${path}flutter';
```
Comment (kevmoo) [inline lib/src/sdk_env.dart]: Why not p.join w/ the path package?

I avoid path concat w/ strings like the plague

File: lib/src/sdk_env.dart [hunk 3]
```diff
@@ -214,3 +235,14 @@ final _versionDownloadRexexp =
     new RegExp(r"^MSG : (?:Downloading |Already cached )([\w-]+) (.+)$");
 
 const _pubEnvironmentKey = 'PUB_ENVIRONMENT';
+
+String _prepareSdkPath(String sdkDir) {
+  String path = '';
+  if (sdkDir != null) {
+    path = sdkDir;
+    if (!sdkDir.endsWith(Platform.pathSeparator)) {
+      path += Platform.pathSeparator;
```
Comment (kevmoo) [inline lib/src/sdk_env.dart]: See comment above WRT pkg/path – avoid string manipulation of paths...

File: lib/src/sdk_env.dart [hunk 4]
```diff
@@ -114,29 +143,14 @@ class PubEnvironment {
...
-      result = await runProc(
-        cmd,
-        args,
-        workingDirectory: packageDir,
-        environment: _environment,
-      );
+      if (isFlutter) {
+        result = await flutterSdk._execUpgrade(packageDir, _environment);
+      } else {
+        result = await _execUpgrade(packageDir);
```
Comment (kevmoo) [inline lib/src/sdk_env.dart]: wondering if this _execUpgrade should be moved to the `DartSdk` class?

Comment (isoos) [inline lib/src/sdk_env.dart]: good catch!

Comment (kevmoo) [inline lib/src/sdk_env.dart]: And then should we have an abstract base class that ...blah blah blah...

Comment (isoos) [inline lib/src/sdk_env.dart]: I don't think that would make it cleaner.

Comment (kevmoo) [inline lib/src/sdk_env.dart]: Just being an astronaut – no worries.

**Comment by kevmoo** [comment]
Ship it!

**Pull request merged**